Refactor Windows and Linux samples into a single codebase.

ryanwinter opened the pull request.

ryanwinter committed:
move samples to use cmake

ryanwinter committed:
testing collapse

ryanwinter committed:
remove collapse

ryanwinter committed:
readme update

ryanwinter committed:
remove linux from titles

ryanwinter committed:
add copyright header to python

ryanwinter committed:
add sample descrption

ryanwinter committed:
Merge branch 'master' of https://github.com/Azure-Samples/IoTMQTTSample

wduraes commented on README.md:
```diff
@@ -45,7 +44,7 @@ The following root certificates are provided in the repository root:
 | Certificate | Path | Description |
 |-|-|-|
-| [Baltimore CyberTrust Root](https://www.digicert.com/kb/digicert-root-certificates.htm) | IoTHubRootCA_Baltimore.pem | Current Root CA set to **expire in 2025** |
+| [Baltimore CyberTrust Root](https://www.digicert.com/kb/digicert-root-certificates.htm) | BaltimoreCyberTrustRoot.crt.pem | Current Root CA set to **expire in 2025** |
```
```suggestion
| [Baltimore CyberTrust Root](https://www.digicert.com/kb/digicert-root-certificates.htm) | BaltimoreCyberTrustRoot.crt.pem | Current Root CA set to **expire in 2025** | **NOT recommended for use** since [IoT Hub migration out of Baltimore](https://techcommunity.microsoft.com/t5/internet-of-things-blog/azure-iot-tls-critical-changes-are-almost-here-and-why-you/ba-p/2393169) is set to start on Feb 2023.
```

wduraes commented on mosquitto/readme.md:
````diff
@@ -0,0 +1,133 @@
+   ```c
+   #define IOTHUBNAME      "{iothub_name}"
+   #define DEVICEID        "{device_id}"
+   #define SAS_TOKEN       "{sas_token}"
````
can we include a sample of what are we expecting here? a.k.a. Iot Hub name: is it just the name, or the FQDN?   also, SAS Toker: just the key or the whole thing?

ryanwinter committed:
readme updates

ryanwinter committed:
Use combined certificate
Use absolute cert path for mosquitto client lib
Use IoT Hub instead of hostname consistently across samples
Use sas token instead of sas key in python

ryanwinter committed:
remove old .gitignore lines
